x/auction/keeper/rewards.go

```diff
@@ -18,11 +18,11 @@ import (
 
 func (k Keeper) FinalizeRewardsAuction() error {
 	now := k.ctx.BlockTime()
-	a, id := k.getRewardsAuction(0)
+	a, id := k.getRewardsAuction(k.currentRewardsAuctionID())
 	if a == nil {
 		return k.initNewAuction(id+1, []sdk.Coin{})
 	}
-	if !a.EndsAt.After(now) {
+	if a.EndsAt.After(now) {
```

Comment:
we need collect rewards after the auction complete

Reply:
No, this is good. The finalize auction does 2 things:
1. settles the current auction
2. collects fees for the next auction
This should be done only AFTER the current auction.

Reply:
in every end block of auction module, it is creating new auction record